src/index.ts

```diff
@@ -1,47 +1,54 @@
 import ejs from "ejs";
-import { getContext, helpers, Property } from "@dylibso/xtp-bindgen";
+import { helpers, getContext, ObjectType, EnumType, ArrayType, XtpNormalizedType, MapType, Parameter, Property } from "@dylibso/xtp-bindgen"
 
-function toGolangType(property: Property): string {
-  if (property.$ref) return goName(property.$ref.name);
-  switch (property.type) {
-    case "string":
-      if (property.format === "date-time") {
-        return "time.Time";
-      }
-      return "string";
-    case "number":
-      if (property.format === "float") {
-        return "float32";
-      }
-      if (property.format === "double") {
-        return "float64";
+function toGolangTypeX(type: XtpNormalizedType, required?: boolean): string {
+  // annotate with null if nullable 
+  const pointerify = (t: string) => {
+    // TODO key off the `type.kind` here instead
+    if (t.startsWith("[]") || t.startsWith("map[")) return t
+    return `${type.nullable || !required ? '*' : ''}${t}`
+  }
+
+  switch (type.kind) {
+    case 'string':
+      return pointerify('string')
+    case 'int32':
+      return pointerify('int32')
+    case 'float':
+      return pointerify('float32')
+    case 'double':
+      return pointerify('float64')
+    case 'byte':
+      return pointerify('byte')
+    case 'date-time':
+      return pointerify('time.Time')
+    case 'boolean':
+      return pointerify('bool')
+    case 'array':
+      const arrayType = type as ArrayType
+      return pointerify(`[]${toGolangTypeX(arrayType.elementType)}`)
+    case 'buffer':
+      return pointerify('[]byte')
+    case 'object':
+      const oType = (type as ObjectType)
+      if (oType.properties?.length > 0) {
+        return pointerify(goName(oType.name))
+      } else {
+        // TODO: shouldn't this be `interface{}` ?
+        return pointerify("map[string]interface{}")
```

Comment:
Now that I think about it, i think this makes sense why it's map[string]interface{} since we're assuming it's coming from json

Reply:
Yep -- I went back and forth if this should be `interface{}` only, or `map[string]interface{}`... for JSON the latter is fine. If we get into more complex encoding, where I could keep a struct a struct, then `interface{}` might be needed, but then we probably wouldn't call it `"object"` in XTP Schema? Unsure.

Reply:
Yeah, that’s what MapType and additionalProperties are for. If I am interpreting correctly

Reply:
I see, then IMO it kind of comes down to flexibility vs. ergonomics. 

On the flexibility side, `interface{}` supports `map[string]interface{}` too, via type-assertion. It just also can be a whole `struct` or any other Go type. So, we get a lot out of it. 

On the ergonomics side, `interface{}` would further need to be type-asserted to `map[string]interface{}` before you can access any field. `map[string]interface{}` only needs the type assertion after you check if the field exists and need to know the type in the key. 

There's the option to do this too, which might be the way to go: 
1. if schema's `object` `contentType` is `application/json`, then do `map[string]interface{}` (as it can't be anything else). 
2. otherwise return `interface{}`. idk if we check if binary is used and override this decision to use `[]byte`, but its not a big leap to go `b, ok := T.([]byte)` and get there from `interface{}`

what do you you think?

Reply:
I think that makes sense. but we don't really know if the object is json encoded or not. in theory it could be multiple encodings or no encodings. I think we want to keep too much about the encoding from leaking into the data layer (even though obviously in languages like go we need to at least annotate with json tags).

I think for this reason it might make sense to make this the most flexible option and make it `interface{}`.

Reply:
wait, we don't know the it's JSON? how is that?

Reply:
we know the thing that contains this is JSON though, right? 

I didn't mean to tie the object and the contentType together in that previous comment

Reply:
>  wait, we don't know the it's JSON? how is that?

The encoding was moved to the parameter. so we're just assuming it's json. 

> if schema's object contentType is application/json, then do map[string]interface{} (as it can't be anything else).

there isn't a contentType on the schema itself. it's on the parameter.

Reply:
yea, actually, interface{} does make this way more flexible - and supports unknown structs that the Go plugin code can freely type assert into. and if it's just an assert to the map version then it's not a lot of code for a plugin dev to write. 

let's switch to `interface{}`